humility-core/src/core.rs

```diff
@@ -985,12 +1075,15 @@ impl Core for GDBCore {
     }
 
     fn read_word_32(&mut self, addr: u32) -> Result<u32> {
-        self.send_32(&format!("m{:x},4", addr))
+        let mut data = [0; 4];
+        self.read_8(addr, &mut data)?;
```

Comment:
Why is this falling back to byte-by-byte read?

Reply:
That is basically what `send_32` did before.  With gdb, whether I request a full word, or 4 bytes, I have to send and receive the same number of bytes, but they would have to be parsed differently.  I think it makes sense to just reuse the byte code.

Reply:
Ack